include/mrdocs/Generator.hpp

```diff
@@ -100,6 +108,33 @@ class MRDOCS_VISIBLE
         std::string_view outputPath,
         Corpus const& corpus) const;
 
+    /** Build reference documentation for the corpus.
+
+        This function invokes the generator to emit
+        the full documentation to an output stream,
+        as a single entity.
+
+        @par Thread Safety
+        @li Different `corpus` object: may be called concurrently.
+        @li Same `corpus` object: may not be called concurrently.
+
+        @return The error, if any occurred.
+
+        @param os The stream to write to.
+
+        @param corpus The metadata to emit.
+
+        @param context The context to use for the output.
+    */
+    MRDOCS_DECL
+    virtual
+    Error
+    buildOne(
+        std::ostream& os,
+        Corpus const& corpus,
+        GenerationContext const& context) const = 0;        
```

Comment:
My impression is that this solution expands the problem rather than solves it. Now, not only we expect a `std::string tagFileName;` (which breaks the contract of the class) but we expect a redundant `std::ostream& tagFileOs;` that needs to be handled and maintained.

I don't know if you're familiar with SOLID, but maybe have a look at https://medium.com/@oleksandra_shershen/solid-principles-implementation-and-examples-in-c-99f0d7e3e868. 

The original problem is that expecting `std::string tagFileName` as a parameter breaks the contract of the parent class because the single responsibility of `buildOne` should be streaming the documentation results to any output. In other words, ideally, the parent `Generator` class shouldn't mention anything related to tagfiles because being able to generate tagfiles is not a property of all generators. If only some generators have this functionality, ideally, there should be an intermediary class (say `GeneratorThatAlsoGeneratesTagfiles` to avoid bikeshedding for now) between `Generator` and the generators that so have tagfiles functionality (`AdocGenerator` and `HTMLGenerator`). "`GeneratorThatAlsoGeneratesTagfiles`" would inherit from `Generator` and define the common API for generating tagfiles (which might have many similarities for all generators). `AdocGenerator` and `HTMLGenerator` would then inherit from "`GeneratorThatAlsoGeneratesTagfiles`" and only implement whatever is different for Adoc and HTML, which I don't expect to be much because only the file extension of the filenames should be different.

The solution I proposed is a `buildTagfile(std::ostream& os, Corpus const& corpus)`, ideally in this new intermediary base class with the common logic for tagfiles. If we really don't want an intermediary class for "convenience", then this `buildTagfile(std::ostream& os, Corpus const& corpus)` function could go directly in the generator and do nothing when the Generator cannot generate tagfiles.

If the logic for tag files can't be decoupled from the logic for `buildOne`, maybe we could consider more complex solutions because it's impossible to implement its single responsibility. But it's clear that tagfiles can be implemented separately because they're basically an XML generator with a different schema.